lhotse/dataset/surt.py

```diff
@@ -0,0 +1,353 @@
+from collections import defaultdict
+from typing import Callable, Dict, List, Union
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+from lhotse import validate
+from lhotse.cut import CutSet
+from lhotse.dataset.input_strategies import BatchIO, PrecomputedFeatures
+from lhotse.utils import LOG_EPSILON, compute_num_frames, ifnone
+from lhotse.workarounds import Hdf5MemoryIssueFix
+
+
+class K2SurtDataset(torch.utils.data.Dataset):
+    """
+    The PyTorch Dataset for the multi-talker ASR task using k2 library.
+    We support the modeling framework known as Streaming Unmixing and Recognition
+    Transducer (SURT), as described in [1] and [2], but this dataset can also be
+    used for other multi-talker ASR approaches such as MT-RNNT [3] and SOT [4].
+    See icefall recipe for usage.
+
+    We take a cut containing possibly overlapping speech and split the supervision
+    segments into one of N channels based on their start times (N is provided), known
+    as ``heuristic error assignment training`` (HEAT) [1]. The supervision segments
+    in each channel are then concatenated and used as the supervision for that channel.
+    If we have features for the source cuts, we can also return them for use in masking
+    losses, for instance.
+
+    [1] Lu, L., Kanda, N., Li, J., & Gong, Y. (2021). Streaming end-to-end multi-talker
+    speech recognition. IEEE Signal Processing Letters, 28, 803-807.
+
+    [2] Raj, D., Lu, L., Chen, Z., Gaur, Y., & Li, J. (2022, May). Continuous streaming
+    multi-talker asr with dual-path transducers. In ICASSP 2022-2022 IEEE International
+    Conference on Acoustics, Speech and Signal Processing (ICASSP) (pp. 7317-7321). IEEE.
+
+    [3] Sklyar, I., Piunova, A., Zheng, X., & Liu, Y. (2022, May). Multi-turn RNN-T for
+    streaming recognition of multi-party speech. In ICASSP 2022-2022 IEEE International
+    Conference on Acoustics, Speech and Signal Processing (ICASSP) (pp. 8402-8406). IEEE.
+
+    [4] Kanda, N., Gaur, Y., Wang, X., Meng, Z., & Yoshioka, T. (2020). Serialized output
+    training for end-to-end overlapped speech recognition. arXiv preprint arXiv:2003.12687.
+
+    .. hint:: Training mixtures can be simulated from single-speaker utterances using
+        :class:`~lhotse.workflows.MeetingSimulation` workflow.
+
+    This dataset expects to be queried with lists of cut IDs,
+    for which it loads features and automatically collates/batches them.
+
+    To use it with a PyTorch DataLoader, set ``batch_size=None``
+    and provide a :class:`SimpleCutSampler` sampler.
+
+    Each item in this dataset is a dict of:
+
+    .. code-block::
+
+        {
+            'inputs': float tensor with shape determined by :attr:`input_strategy`:
+                      - single-channel:
+                        - features: (B, T, F)
+                        - audio: (B, T)
+                      - multi-channel: currently not supported
```

Comment:
is it really not supported? the rest of the docs suggests otherwise

Reply:
Here by "channel" we are referring to the audio channel (i.e. number of microphones) in the input recording. In the rest of the code, "channel" actually refers to an output channel on which a transcription is done. For example, overlapping speech may be transcribed simultaneously on 2 output channels.